context-propagation/src/main/java/io/micrometer/context/DefaultContextSnapshot.java

```diff
@@ -77,99 +76,37 @@ public Scope setThreadLocals(Predicate<Object> keyPredicate) {
-            Object key = threadLocalAccessor.key();
-            Object value = ((ContextAccessor<C, ?>) contextAccessor).readValue((C) context, key);
-            if (value != null) {
-                previousValues = setThreadLocal(key, value, threadLocalAccessor, previousValues);
-            }
-        }
-        return DefaultScope.from(previousValues, registry);
-    }
-
-    @SuppressWarnings("unchecked")
-    static <C> Scope setThreadLocalsFrom(Object context, ContextRegistry registry, String... keys) {
-        if (keys == null || keys.length == 0) {
-            throw new IllegalArgumentException("You must provide at least one key when setting thread locals");
-        }
-        ContextAccessor<?, ?> contextAccessor = registry.getContextAccessorForRead(context);
-        Map<Object, Object> previousValues = null;
-        for (String key : keys) {
-            Object value = ((ContextAccessor<C, ?>) contextAccessor).readValue((C) context, key);
-            if (value != null) {
-                for (ThreadLocalAccessor<?> threadLocalAccessor : registry.getThreadLocalAccessors()) {
-                    if (key.equals(threadLocalAccessor.key())) {
-                        previousValues = setThreadLocal(key, value, threadLocalAccessor, previousValues);
-                    }
-                }
-            }
-        }
-        return DefaultScope.from(previousValues, registry);
-    }
-
-    static ContextSnapshot captureAll(ContextRegistry contextRegistry, Predicate<Object> keyPredicate,
-            Object... contexts) {
-
-        DefaultContextSnapshot snapshot = captureFromThreadLocals(keyPredicate, contextRegistry);
-        for (Object context : contexts) {
-            snapshot = captureFromContext(keyPredicate, contextRegistry, snapshot, context);
-        }
-        return (snapshot != null ? snapshot : emptyContextSnapshot);
-    }
-
-    @Nullable
-    private static DefaultContextSnapshot captureFromThreadLocals(Predicate<Object> keyPredicate,
-            ContextRegistry contextRegistry) {
-
-        DefaultContextSnapshot snapshot = null;
-        for (ThreadLocalAccessor<?> accessor : contextRegistry.getThreadLocalAccessors()) {
-            if (keyPredicate.test(accessor.key())) {
-                Object value = accessor.getValue();
-                if (value != null) {
-                    snapshot = (snapshot != null ? snapshot : new DefaultContextSnapshot(contextRegistry));
-                    snapshot.put(accessor.key(), value);
-                }
-            }
-        }
-        return snapshot;
-    }
-
-    @SuppressWarnings("unchecked")
-    static DefaultContextSnapshot captureFromContext(Predicate<Object> keyPredicate, ContextRegistry contextRegistry,
-            @Nullable DefaultContextSnapshot snapshot, Object... contexts) {
+    static <V> Map<Object, Object> setThreadLocal(Object key, V value, ThreadLocalAccessor<?> accessor,
+            @Nullable Map<Object, Object> previousValues) {
```

Comment:
This method could move up one, ahead of `clearThreadLocal` following their order of appearance.